src/invenio-files-js/services/InvenioFilesAPI.js

```diff
@@ -36,9 +36,13 @@ function InvenioFilesAPI($http, Upload) {
     * Start upload
     * @memberof InvenioFilesAPI
     * @param {object} args - The upload parameters.
+    * @param {boolean} multipartUpload - If the upload is multipart.
     */
-  function upload(args) {
-    return Upload.upload(args);
+  function upload(args, multipartUpload) {
+    if (multipartUpload) {
+      return Upload.upload(args);
+    }
+      return Upload.http(args);
```

Comment:
When I do this - it all works:

```javascript
args.data = args.data.file;
return Upload.http(args);
```

I think it's simple the ``data`` attribute in args, which has to have the file directly, and not wrapped.